### ISSUE TITLE
TypeScript Template Repository Build Fail (Node v17.6.0)

### ISSUE THREAD

**Issue by dylhack** [vercel/next.js#35029]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
/bin/sh: 1: pnpm: not found

    Operating System:
      Platform: linux
      Arch: x64
      Version: #202202110435~1644952300~21.10~96763f1 SMP Tue Feb 15 19:52:40 U
    Binaries:
      Node: 17.6.0
      npm: 8.5.1
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.1-canary.6
      react: 17.0.2
      react-dom: 17.0.2

```

### What browser are you using? (if relevant)

Chrome 100.0.4878.0

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

After installing the default template from `npx create-next-app@latest --ts`, attempting to build the project fails.

```sh
next build
```

**Error**
```
12:57:04 dylhack@pop-os test ±|main|→ next build
info  - Checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
[    ] info  - Generating static pages (0/3)
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:18:327)
    at Object.exports.useContext (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:22:269)
    at Head (/home/dylhack/test/node_modules/next/dist/shared/lib/head.js:179:34)
    at d (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
    at bb (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
    at a.b.render (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
    at a.b.read (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
    at Object.exports.renderToString (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
    at Object.renderPage (/usr/lib/node_modules/next/dist/server/render.js:751:45)
    at Object.defaultGetInitialProps (/usr/lib/node_modules/next/dist/server/render.js:389:51)

Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:18:327)
    at Object.exports.useContext (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:22:269)
    at Head (/home/dylhack/test/node_modules/next/dist/shared/lib/head.js:179:34)
    at d (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
    at bb (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
    at a.b.render (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
    at a.b.read (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
    at Object.exports.renderToString (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
    at Object.renderPage (/usr/lib/node_modules/next/dist/server/render.js:751:45)
    at Object.defaultGetInitialProps (/usr/lib/node_modules/next/dist/server/render.js:389:51)

Error occurred prerendering page "/404". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:18:327)
    at Object.exports.useContext (/home/dylhack/test/node_modules/react/cjs/react.production.min.js:22:269)
    at Head (/home/dylhack/test/node_modules/next/dist/shared/lib/head.js:179:34)
    at d (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
    at bb (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
    at a.b.render (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
    at a.b.read (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
    at Object.exports.renderToString (/usr/lib/node_modules/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
    at Object.renderPage (/usr/lib/node_modules/next/dist/server/render.js:751:45)
    at Object.defaultGetInitialProps (/usr/lib/node_modules/next/dist/server/render.js:389:51)
info  - Generating static pages (3/3)

> Build error occurred
Error: Export encountered errors on following paths:
	/
	/_error: /404
	/_error: /500
    at /usr/lib/node_modules/next/dist/export/index.js:498:19
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Span.traceAsyncFn (/usr/lib/node_modules/next/dist/trace/trace.js:75:20)
    at async /usr/lib/node_modules/next/dist/build/index.js:1024:17
    at async Span.traceAsyncFn (/usr/lib/node_modules/next/dist/trace/trace.js:75:20)
    at async /usr/lib/node_modules/next/dist/build/index.js:898:13
    at async Span.traceAsyncFn (/usr/lib/node_modules/next/dist/trace/trace.js:75:20)
    at async Object.build [as default] (/usr/lib/node_modules/next/dist/build/index.js:82:25)
```

### Expected Behavior

The project should build a production version successfully

### To Reproduce

Install Node v17
```
npm i --global next
npx create-next-app@latest --ts
next build
# Error
```

**Comment by dylhack** [comment]
There is additional detail on the Discord in the `default-ts-build-fail` thread.

**Comment by dylhack** [comment]
I was able to reproduce this in Docker using the node image `node:17-alpine3.14`

**Comment by dylhack** [comment]
This error only appears when next is installed globally rather than using the local workspace dependency.

**Comment by balazsorban44** [comment]
Out of curiosity, why would you have `next` installed globally? It does not really make sense and sounds like it can have unforeseen consequences like this.

**Comment by balazsorban44** [comment]
Could you attach your `Dockerfile`? I am unable to reproduce with `next` being globally installed and using Node.js `17.4.0`.

**Comment by balazsorban44** [comment]
So after some internal discussion, we are not recommending `next` being installed globally, as all imports are going to fail in the project, as Node.js isn't able to pull in modules like `next/link` etc, which could result in the above-mentioned errors.

**Comment by dylhack** [comment]
Thanks! It would be much appreciated for a warning in the get started area

**Comment by dylhack** [comment]
> Out of curiosity, why would you have `next` installed globally? It does not really make sense and sounds like it can have unforeseen consequences like this.

I read the npm script that called next directly so instead of doing npm run I decided to go with installing it globally. It didn't click until I realized that it was relying on the workspace-dependant next install rather than any install, with that in mind I attempted to create a Dockerfile around that, and of course, I ran into these errors.

A warning would be nice or a troubleshooting topic.
